addons/web_editor/models/ir_qweb_fields.py

```diff
@@ -51,23 +51,37 @@ def _compile_node(self, el, compile_context, indent):
             while node is not None and nb_real_elements_in_hierarchy < 2:
                 if node.tag != 't' or 't-call' in node.attrib:
                     nb_real_elements_in_hierarchy += 1
+                    sub_call = node.get('t-call')
+                    # If the node is a t-call, we create the snippet-sub-call-key to still be able
+                    # to add the "data-snippet" attrib of the sub called snippet
+                    # Then before the childs node are compiled the t-call attrib will be delete by
+                    # the compilation so we have to register it so a child of a t-call is not
+                    # believed to be top level
+                    if sub_call:
+                        node.set(
+                            't-options',
+                            f"{{'snippet-key': '{snippet_key}', 'snippet-sub-call-key': '{sub_call}'}}"
+                        )
+                        for child in node.iter(tag=etree.Element):
+                            child.set("t_call_child", "true")
+                    if "t_call_child" in node.attrib:
+                        nb_real_elements_in_hierarchy += 1
```

Comment:
If I understand this right you do `nb_real_elements_in_hierarchy += 1` twice for the same node if you land on this line... is this right? :thinking: Also, maybe that `nb_real_elements_in_hierarchy` variable is wrong in the first place and there is a better way to handle this.